scripts/gen_feeds_daily.sh

```diff
@@ -1,14 +1,104 @@
-#!/bin/sh
+#!/usr/bin/env bash
+
+# check we have the environment variables PRODUCT_OPENER_FLAVOR and PRODUCT_OPENER_FLAVOR_SHORT, otherwise exit
+
+if [ -z "$PRODUCT_OPENER_FLAVOR" ] || [ -z "$PRODUCT_OPENER_FLAVOR_SHORT" ]; then
+    >&2 echo "Environment variables PRODUCT_OPENER_FLAVOR and PRODUCT_OPENER_FLAVOR_SHORT are required"
+    exit 1
+fi
+
+# this script must be launched from server root (e.g. /srv/off)
+# check that we have a lib/ProductOpener/Paths.pm file, otherwise exit
+
+if [ ! -f lib/ProductOpener/Paths.pm ]; then
+    >&2 echo "lib/ProductOpener/Paths.pm not found. ./scripts/gen_feeds_daily.sh must be launched from server root"
+    exit 1
+fi  
+
+export PERL5LIB=lib:$PERL5LIB
+
+# load paths
+. <(perl -e 'use ProductOpener::Paths qw/:all/; print base_paths_loading_script()')
+
+# load PRODUCT_OPENER_DOMAIN
+. <(perl -e 'use ProductOpener::Config qw/:all/; print "export PRODUCT_OPENER_DOMAIN=$server_domain\n";')
+
+# we should now have PRODUCT_OPENER_DOMAIN set (from Config.pm in production mode), check it
+if [ -z "$PRODUCT_OPENER_DOMAIN" ]; then
+    >&2 echo "Environment variable PRODUCT_OPENER_DOMAIN not set"
+    exit 1
+fi
+
+cd $OFF_SCRIPTS_DIR
+
+# off-pro flavor: we don't generate most exports
+# but we have some special processing
+if [ "$PRODUCT_OPENER_FLAVOR" == "off-pro" ]; then
+    echo "Generating feeds for off-pro flavor"
+    ./save_org_product_data_daily_off_pro.pl
+    echo "Skipping exports for off-pro flavor"
+    exit 0
+fi
 
-cd /srv/off/scripts
-export PERL5LIB="../lib:${PERL5LIB}"
 
 ./remove_empty_products.pl
-./compute_missions.pl
+./gen_top_tags_per_country.pl
+
+# Generate the CSV and RDF exports
 ./export_database.pl
-./mongodb_dump.sh /srv/off/html openfoodfacts 127.0.0.1 off
 
-cd /srv/off/html/data
-gzip < en.openfoodfacts.org.products.rdf > en.openfoodfacts.org.products.rdf.gz
-gzip < fr.openfoodfacts.org.products.rdf > fr.openfoodfacts.org.products.rdf.gz
+cd $OFF_PUBLIC_DATA_DIR
+for export in en.$PRODUCT_OPENER_DOMAIN.products.csv fr.$PRODUCT_OPENER_DOMAIN.products.csv en.$PRODUCT_OPENER_DOMAIN.products.rdf fr.$PRODUCT_OPENER_DOMAIN.products.rdf; do
+   nice pigz < $export > new.$export.gz
+   mv -f new.$export.gz $export.gz
+done
+
+# Copy CSV and RDF files to AWS S3 using MinIO client
+mc cp \
+    en.$PRODUCT_OPENER_DOMAIN.products.csv \
+    en.$PRODUCT_OPENER_DOMAIN.products.csv.gz \
+    en.$PRODUCT_OPENER_DOMAIN.products.rdf \
+    fr.$PRODUCT_OPENER_DOMAIN.products.csv \
+    fr.$PRODUCT_OPENER_DOMAIN.products.csv.gz \
+    fr.$PRODUCT_OPENER_DOMAIN.products.rdf \
+    s3/openfoodfacts-ds
+
+# Generate the MongoDB dumps and jsonl export
+cd $OFF_SCRIPTS_DIR
+
+./mongodb_dump.sh $OFF_PUBLIC_DATA_DIR $PRODUCT_OPENER_FLAVOR 10.1.0.102 $PRODUCT_OPENER_FLAVOR_SHORT
```

Comment:
nit: could we use mongodb address from config here, instead of hardcoded ?

Reply:
good idea, I will do that